baybe/simulation/_imputation.py

```diff
@@ -0,0 +1,87 @@
+"""Target value imputation functionality."""
+
+from typing import Literal
+
+import numpy as np
+import pandas as pd
+
+from baybe.targets import NumericalTarget
+from baybe.targets.enum import TargetMode
+
+
+def _impute_lookup(
```

Comment:
Making it private? Perhaps you were looking at the wrong spot, but I didn't change anything here, the function was already private before. On the contrary, I agree with what you wrote, that's why I made `_look_up_target_values` public and renamed it to `look_up_targets`, as the high-level user-facing entry point that now simply acts as a selector routine – `_impute_lookup` is one of the internal choices that will become active when a user passes a dataframe as lookup mechanism. Does this make sense to you?

Reply:
if you think it should be private (prob reasonable for consistencty and because its exposed as part of another function), I was just questioning why but have no strong opinion

Reply:
For now, I'd say let's simply keep it private since – as you said – we have that dispatcher public that will invoke it upon request. That makes one function less to be maintained as public. Step from private to public is easy and we can always do it when we think it'd be handy 🙃